doc/design/cli.md

```diff
@@ -0,0 +1,105 @@
+# SQLFlow Command Line Interface
+
+## Overview
+
+At the moment, users can build SQLFlow into a server or a [command line binary](../run/repl.md). Both of the executables have all the functionalities of SQLFlow, and both of them can only run in a Docker container.
+
+For a deployment of SQLFlow in a production environment, the SQLFlow server runs in a Kubernetes cluster. Users access SQLFlow from a Web IDE or the Jupyter Notebook.
+
+For an individual developer, one may deploy SQLFlow on her personal computer as described in [Run SQLFlow Using Docker](../run/docker.md)
+
+For the above two typical scenarios, SQLFlow lacks an easy-to-use user interface for quick start, quick verification, or quick deployment. Jupyter Notebooks and web IDEs still have a certain learning curve that reduces the user experience of SQL programmers. 
+
+From another point of view, a CLI could overtake a GUI in user experience for a simple language as SQLFlow: it's easier to design, easier to implement, easier to obtain. For example, repetitive tasks can be simplified by line editing and history mechanisms. With a CLI program, a user can easily access all the power of SQLFlow from a terminal as:
+
+```bash
+$ sqlflow
+sqlflow> _
+```
+
+To achieve this goal, we also want to hide implementation details about Docker because users don't have to know this. As a result, there are two possible solutions:
+1. Wrap the current SQLFlow REPL in a script, we already have such an [attempt](https://github.com/sql-machine-learning/sqlflow/pull/2114).
+1. Decouple the REPL implementation from the core code of SQLFlow and provide a native binary that only functions as a pure client.
+
+We decide to choose the second solution because it's more lightweight, if a user already has a SQLFlow deployment in her company, all she has to do is to download the native binary.
+
+## Design
+
+### Naming
+
+`REPL` is not a good name, We'll rename the binary as simply `sqlflow`.
```

Comment:
Currently, the command-line binary form of SQLFlow has the name `repl`, which is the name of a user interface design philosophy.  Many command-line tools, including `mysql` and `python`, implement the UI of REPL.  Let us follow the convention of `mysql` and `mysqld` to name the new client `sqlflow`.

Reply:
Done.